tools/cli/src/options/loadOptions.ts

```diff
@@ -13,33 +13,26 @@ export function loadOptionsForCommand(path?: string) {
         key: K,
         defaults?: Partial<CommandCliOptions[K]>,
     ): ResultPromise<Partial<CommandCliOptions[K]>, string> {
-        if (readOptions) {
-            return Ok(joinOptions(key, defaults)(readOptions));
-        }
-
-        const readResult = await readFile(path ?? DEFAULT_OPTIONS_FILENAME);
-        if (readResult.isErr()) {
-            return path ? readResult : Ok({});
-        }
-
-        return AsyncResult.equip(readResult)
-            .then(parseJSONC<CliOptions>)
-            .then((opts) => (readOptions = opts))
-            .then(joinOptions(key, defaults)).inner;
+        return pipe(
+            path,
+            (path) => (!readOptions ? loadOptions(path) : Ok(readOptions)),
+            R.map((data) => (data ? (readOptions = data) : data)),
```

Comment:
how it works - you're assigning something when data is defined, but you're not returning it and in the next method you're trying to use it. Are you sure it works correctly? I think the result will be undefined here.

Reply:
`(readOptions = data)` returns `data`:
![image](https://github.com/backtrace-labs/backtrace-javascript/assets/9085864/0c3759bb-df7e-4e80-ad2e-837e070624df)